lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -524,6 +524,19 @@ ChangeResult TypeAnalyzer::visitOperation(
     return incorporateKnowledge(op->getResult(0), knowledge);
   }
 
+  // Dtype is always float32, except for float64.
+  if (isa<AtenTanhOp, AtenExpOp, AtenSinOp, AtenCosOp,
+          AtenSigmoidOp, AtenReciprocalOp, AtenLogOp, AtenSqrtOp,
+          AtenLog2Op, AtenRsqrtOp, AtenErfOp>(op)) {
+    ValueKnowledge knowledge =
+        ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+    knowledge.dtype = Float32Type::get(op->getContext());
+    if (operands[0]->getValue().dtype == Float64Type::get(op->getContext())) {
```

Comment:
nit: you can do `operands[0]->getValue().dtype.isa<Float64Type>()`

Reply:
Got it. Thank you!